providers/zohocrm/errors.go

```diff
@@ -0,0 +1,45 @@
+package zohocrm
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+/*
+The normal standard response from ZohoCRM look like this:
+{
+	"data": [{...}, {...}],
+	"info": {...}
+}
+*/
+
+// responseHandler wraps the http.StatusNotModified response into http.StatusOK
+// and sends back an empty data field response.
+func responseHandler(resp *http.Response) (*http.Response, error) { //nolint:cyclop
+	// responseData represents data we will send back when the response status code 304.
+	responseData := map[string]any{
+		"data": []any{},
+		"info": map[string]any{
+			"page":         1,
+			"more_records": false,
+		},
+	}
+
+	// When the ZohoCRM API responds with 304 status code,
+	// this indicates there is no data modified since the modification time provided.
+	// We modify the response to return 200, And an empty data field response.
```

Comment:
let's change the comment as below

```
// When there is no new record after the specified time `since`, ZohoCRM returns `304 Status Not Modified`.  Then we wrap this response to 200 Status Okay with empty array data. 
```